tests/contract/search_concepts_contract_test.py

```diff
@@ -0,0 +1,136 @@
+import json
+import re
+import time
+from datetime import datetime
+
+import pytest
+import requests
+from jsonpath_ng import parse
+from requests import get
+from urllib3.exceptions import MaxRetryError, NewConnectionError
+
+from tests.contract.contract_utils import expected_page_keys
+from tests.contract.search_all_contract_test import service_url
+
+
+@pytest.fixture(scope="function")
+def wait_for_concepts():
+    timeout = time.time() + 90
+    try:
+        while True:
+            response = get("http://localhost:8000/indices?name=concepts")
+            if response.json()[0]['count'] >= 530:
+                break
+            if time.time() > timeout:
+                pytest.fail(
+                    'Test function setup: timed out while waiting for fulltext-search, last response '
+                    'was {0}'.format(response.json()["count"]))
+            time.sleep(1)
+    except (requests.exceptions.ConnectionError, ConnectionRefusedError, MaxRetryError, NewConnectionError):
+        pytest.fail('Test function setup: could not contact fdk-fulltext-search container')
+    yield
+
+
+indices_name = "concepts"
+data_type = "concept"
+concept_url = service_url + f"/{indices_name}"
+
+
+class TestConceptSearch:
+
+    @pytest.mark.contract
+    def test_response_should_have_correct_content(self, api, wait_for_concepts):
+        result = requests.post(concept_url)
+        result_json = result.json()
+        content_keys = result_json.keys()
+        assert result.status_code == 200
+        assert "hits" in content_keys
+        result_data_types = [match.value for match in parse("hits[*].type").find(result_json["hits"])]
+        for dt in result_data_types:
+            assert dt == data_type
+        assert "page" in content_keys
+        page_key_matches = list(set(result_json["page"].keys()) & set(expected_page_keys))
+        assert page_key_matches.__len__() == expected_page_keys.__len__()
+        assert "aggregations" in content_keys
+        agg_keys = result_json["aggregations"].keys()
+        assert "los" in agg_keys
+        assert "orgPath" in agg_keys
+
+    @pytest.mark.contract
+    def test_hits_should_be_correctly_sorted_on_title(self, api, wait_for_concepts):
+        """
+            1. exact match
+            2. word in title
+        """
+        search_str = "RA-0554 Pris på juridisk tjenesteyting"
+        body = {
```

Comment:
Bytt "RA-0554 Pris på juridisk tjenesteyting" med en verdi som har esakt match i tittel for begrep (RA-0554 er vel en informasjonsmodell)? Og legg til en test på at resultatet i det hele tatt inneholder esakt match(mistenker at testen kan gå igjennom om det ikke finnes noen form fo rmatch i tittel)

Reply:
Endrer search_str og legger til at "exact_match_exists" må være True